src/components/badge/index.js

```diff
@@ -236,7 +237,7 @@ export default class Badge extends PureBaseComponent {
           activeOpacity={activeOpacity}
           {...animationProps}
         >
-          {icon ? this.renderIcon() : this.renderLabel()}
+          {icon ? this.renderIcon() : label ? this.renderLabel() : undefined}
```

Comment:
regarding this `label ? this.renderLabel() : undefined`
why shouldn't you move this logic to be part of `renderLabel` 
ternary inside a ternary is never readable...

Reply:
Done